assembl/static2/js/app/components/login/sendPwdForm.jsx

```diff
@@ -1,25 +1,29 @@
+// @flow
 import React from 'react';
-import { connect } from 'react-redux';
 import { Translate, I18n } from 'react-redux-i18n';
 import { form, FormGroup, FormControl, Button } from 'react-bootstrap';
+
 import { getDiscussionSlug } from '../../utils/globalFunctions';
-import { requestPasswordChangeAction } from '../../actions/authenticationActions';
-import inputHandler from '../../utils/inputHandler';
 import { displayAlert } from '../../utils/utilityManager';
+import { type ChangePasswordResponse } from '../../pages/requestPasswordChange';
 
-class SendPwdForm extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { identifier: null };
-    this.submitHandler = this.submitHandler.bind(this);
-    this.handleInput = this.handleInput.bind(this);
-  }
+type Props = {
+  passwordChangeResponse: ChangePasswordResponse,
+  requestPasswordChange: (string, string) => void
+};
+
+type State = {
+  identifier: string | null
+};
 
-  componentWillReceiveProps(nextProps) {
-    const { auth } = nextProps;
-    const resp = auth.passwordChangeRequest;
-    if (resp && resp.success === false) {
-      const firstError = resp.data[0];
+class RequestNewPasswordForm extends React.Component<Props, State> {
+  state = { identifier: null };
+
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
componentWillReceiveProps? https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Reply:
I will not change this in this PR